pkg/controllers/clusters/clusterstatus/clusterstatus_controller_test.go

```diff
@@ -127,3 +130,149 @@ func (suite *StatusSuite) TestAggregateLimitedSingleLabels() {
 		},
 		commonLabels, "failed to aggregate single node")
 }
+
+func TestAggregateLimitedLabels(t *testing.T) {
+	type arg struct {
+		nodes         []*corev1.Node
+		threshold     float32
+		labelPrefixes []string
+	}
+	var tests = []struct {
+		name string
+		args arg
+		want map[string]string
+	}{
+		{
+			name: "aggregate one",
+			args: arg{
+				nodes: []*corev1.Node{
+					{
```

Comment:
We could use function `BuildNode` to improve the code quality.

Reply:
OK